habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py

```diff
@@ -290,9 +485,40 @@ def get_all_scenedataset_receptacles(sim) -> Dict[str, Dict[str, List[str]]]:
     return receptacles
 
 
+def import_tri_mesh_ply(ply_file: str) -> Tuple[List[mn.Vector3], List[int]]:
+    """
+    Returns a Tuple of (verts,indices) from a ply mesh using magnum trade importer.
+
+    :param ply_file: The input PLY mesh file. NOTE: must contain only triangles.
+    """
+    manager = mn.trade.ImporterManager()
+    # TODO: replace AssimpImporter once a better importer can handle binary PLY or we use another format
+    importer = manager.load_and_instantiate("AssimpImporter")
+    importer.open_file(ply_file)
+
+    # NOTE: We don't support mesh merging or multi-mesh parsing currently
+    if importer.mesh_count > 1:
+        raise NotImplementedError("TODO: multi-mesh receptacle support.")
+
+    mesh_ix = 0
+    mesh = importer.mesh(mesh_ix)
+    assert (
+        mesh.primitive == mn.MeshPrimitive.TRIANGLES
+    ), "Must be a triangle mesh."
+
+    # zero-copy reference to importer datastructures
+    mesh_data: Tuple[List[mn.Vector3], List[int]] = (
```

Comment:
If you'd want to be *exact* with the typing annotation, the actual type is [cr.containers.StridedArrayView1D](https://doc.magnum.graphics/python/corrade/containers/StridedArrayView1D/), not a `List`, but right now I don't have the `[]` "templating" implemented so saying `cr.StridedArrayView1D[mn.Vector3]` would probably cause Python to complain.... and without it it's maybe too nondescriptive (like `np.ndarray` is). I'll investigate what can be done on that front (implementing `__getitem__` on the typeclass? huh).